File: JAVA/3075. Maximize Happiness of Selected Children.java
import java.util.Arrays;
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {        
        int n = happiness.length;
        Arrays.sort(happiness);
        int count = 0;
        long sum = 0;
        for (int i = n - 1; count < k; --i){
            sum += Math.max(happiness[i] - count, 0);
            ++count;
        }
        return sum;
    }
}

File: C++/3075. Maximize Happiness of Selected Children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end());
        int d = 0;
        int i = happiness.size() - 1;
        long long ans = 0;
        while(k > 0 && i >= 0){
            if(happiness[i] - d > 0){
                ans += (long long)(happiness[i] - d);
            }
            k--;
            i--;
            d++;
        }
        return ans;
    }
};
